source/whats-new.txt

```diff
@@ -66,7 +66,27 @@ The 5.0 driver release introduces the following features:
 
 - Changes to the ``ConnectionId`` class. To learn more, see the
   :ref:`Version 5.0 Breaking Changes <java-breaking-changes-v5.0>`.
-
+  
+- Changes to the ``com.mongodb.client.MongoDatabase.listCollectionNames()``
+  methods. The return type is now a
+  ``com.mongodb.client.ListCollectionNamesIterable`` type, while previously it
+  was a ``MongoIterable<String>``. This changes allows the return value to be
+  configured using the ``ListCollectionNamesIterable.authorizedCollections()``
+  method and specifying the ``authorizedCollections`` option. Equivalent changes
+  were made to following classes and interfaces:
+
+  - ``com.mongodb.reactivestreams.client.MongoDatabase``
+  - ``org.mongodb.scala.MongoDatabase``
+  - ``com.mongodb.kotlin.client.MongoDatabase``
+  - ``com.mongodb.kotlin.client.coroutine.MongoDatabase``
+
+  These changes introduce a :ref:` binary breaking change to the listCollections
+  command<java-breaking-changes-v5.0-list-collections>`. For more information
```

Comment:
As described in https://github.com/mongodb/docs-java/pull/516#discussion_r1473806252, the binary breaking change was made to the `listCollectionNames` methods, not the `listCollections` command.